.github/workflows/lint.yaml

```diff
@@ -31,11 +31,11 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
         with:
-          fetch-depth: 0
+          fetch-depth: 2
       - name: Run pre-commit
         run: |
           pre-commit install --hook-type pre-commit --hook-type pre-push
-          pre-commit run --show-diff-on-failure --color=always --from-ref ${{ github.event.pull_request.base.sha && github.event.pull_request.base.sha || github.event.before }} --to-ref HEAD
+          pre-commit run --show-diff-on-failure --color=always --from-ref HEAD~1 --to-ref HEAD
```

Comment:
I don't this is correct if a PR has multiple commits in it

Reply:
local workspace has exactly 2 commits: a HEAD commit from the target branch, and a merge commit (HEAD from target branch + PR changes)

Reply:
Before making this PR I tested this in a sandbox repo.  Looks like actions/checkout merges PR into HEAD (target branch) as part of a checkout.  Here I added `git log` in precommit step, you can see a merge commit: https://github.com/youtube/cobalt_sandbox/actions/runs/16946414577/job/48028407363?pr=793

Effectively using `fetch-depth:2` guarantees to have HEAD being merge commit, and HEAD~1 being current HEAD from the target branch.

Reply:
>Just refreshed and saw your most recent comment. Sg in that case, maybe add a comment that this is dependent on the implementation of actions/checkout

Added.